### PR TITLE
BUG: retry counter

### PR DESCRIPTION
**Pull request by erfannariman**
Looks like the retry counter happens after the login, which is failing, so counter will never increase.

### CODE REVIEW

**Comment by TimvdHeijden** [comment]
@erfannariman Think my last addition at least prevents the login request from going into the loop.

File: twinfield/login.py [hunk 1]
```diff
@@ -77,15 +77,14 @@ def do_request(self, url: str, headers: dict, data=None, req_type: str = "POST")
...
         Returns
         -------
         output
             The response of the API request
         """
         success = False
-        retry = 1
-
+        retry = 0
+        login_request = True if "https://login.twinfield.com/auth/authentication/connect" in url else False
```
Comment (erfannariman) [inline twinfield/login.py:87]: Can you simplify this line with:

```
"https://login.twinfield.com/auth/authentication/connect" in url
```

File: twinfield/login.py [hunk 2]
```diff
@@ -101,12 +100,13 @@ def do_request(self, url: str, headers: dict, data=None, req_type: str = "POST")
...
+                retry += 1
                 logging.exception(
                     f"No response or error, retrying in {self.sec_wait} seconds. "
                     f"Retry number: {retry}. Error message: {e}"
                 )
                 time.sleep(self.sec_wait)
-                # failed request, retry with new login.
-                self.cluster = self.determine_cluster()
-                retry += 1
+                # failed request, retry with new login. DO not do whne making login requests
```
Comment (erfannariman) [inline twinfield/login.py:109]: Two typo's `Do` and `when`

**Review by erfannariman — COMMENTED** [review]
Looks good, two small changes @TimvdHeijden

File: twinfield/login.py (hunk 1 above)
Comment (TimvdHeijden) [reply twinfield/login.py]: Done

File: twinfield/login.py (hunk 2 above)
Comment (TimvdHeijden) [reply twinfield/login.py]: Done

**Comment by erfannariman** [comment]
Looks good, you can merge this and make a new release @TimvdHeijden

**Pull request merged**